f/foldr.xsl

```diff
@@ -10,7 +10,7 @@
       <xsl:param name="pList" select="/.."/>
       
       <xsl:choose>
-         <xsl:when test="not($pList)">
```

Comment:
Not that I object, but I am intrigued.
Could you, please provide a complete example where the evaluation of `not($pList)` produces a different value from that of `empty($pList)` ?

Also, why two PRs for the same issue?

Reply:
@dnovatchev Let me reply to your questions in reverse order:

 2- This is because I'm a rank amateur at GitHub (I've never even tried the command-line version); and have also forgotten half of what I knew about it two years back when I was not quite so rank. I do apologize, and am trying to recover my former skills :)

 1- Trying out FXSL for the first time (and very excited about it), I had several years' annual membership-rosters (XML, each) for an organization, and wanted to merge them into one list, in a specific way (e.g., maintaining data on which years each person was a member). My callback template did the merging Each year's roster was in a separate sub-folder, named by the year (e.g., 2020).

Naturally, my list of "values" to fold was a sequence of year-numbers. I tried it out first on a small range -- only three years. To my surprise, I got this error message: "Effective boolean value is not defined for a sequence of two or more items starting with a numeric value" -- and clicking on that message took me to the line in foldl.xsl with this not() function-call.

Digging into this, I discovered that the changes in conversion-to-boolean rules (in 2.0) made it such that  if you try to coerce to boolean a sequence of two or more integers, for example, it will result in an error, instead of a 'true' value. Michael K. says (on p. 552 of 4th ed. of XSLT & XPath 2.0):

"Note that [in XPath 2.0] the effective boolean value of a sequence doesn't simply test whether the sequence is empty, because of the special cases for a singleton sequence. If you want to test whether a sequence is empty, use the empty() or exists() functions ..."

Thank you for accepting the (reduplicated) change-suggestionss, and for a marvelous XSLT library!

Reply:
**I am glad that someone serious is using FXSL** !  :)

Do you also know/use XPath 3.1 (maps, arrays, the arrow operator)?

What about a serious functional-programming language like Haskell?

Any better ways to communicate with you, besides github comments? Are you a member of the XML.com chat?

https://www.xml.com/news/2020-04-slack-workspace-for-the-xml-community/

https://join.slack.com/t/xmlcom/shared_invite/zt-ebhux9sm-bkyxD0alWjgtvcE8zzcHZQ

Reply:
Great - thanks for the invite! When I get more comfortable with Slack, I'll answer those questions :D